### PR TITLE
708 requisition database schema

### LINKED ISSUE #708: Requisition DB Schema
https://github.com/openmsupply/remote-server/issues/703

### PR DESCRIPTION
**Pull request by andreievg**
Closes #708 

I've deleted repository tests, will reinstate

### CODE REVIEW

File: repository/migrations/postgres/2021-09-17T10:00_create_requisition_table/up.sql [hunk 1]
```diff
@@ -0,0 +1,23 @@
+-- Create requisition table.
+
+CREATE TYPE requisition_type AS ENUM ('REQUEST', 'RESPONSE');
+CREATE TYPE requisition_status AS ENUM ('DRAFT', 'NEW', 'SENT', 'FINALISED');
+
+CREATE TABLE requisition (
+    id TEXT NOT NULL PRIMARY KEY,
+    requisition_number BIGINT NOT NULL,
+    store_id TEXT NOT NULL REFERENCES name(id),
```
Comment (jmbrunskill) [inline repository/migrations/postgres/2021-09-17T10:00_create_requisition_table/up.sql:9]: Looks like name and store might be the wrong way around in this sql?

```suggestion
    store_id TEXT NOT NULL REFERENCES store(id),
    name_id TEXT NOT NULL REFERENCES name(id),
```

File: repository/migrations/sqlite/2021-09-17T10:00_create_requisition_table/up.sql [hunk 2]
```diff
@@ -0,0 +1,20 @@
+-- Create requisition table.
+
+CREATE TABLE requisition (
+    id TEXT NOT NULL PRIMARY KEY,
+    requisition_number BIGINT NOT NULL,
+    store_id TEXT NOT NULL REFERENCES name(id),
```
Comment (jmbrunskill) [inline repository/migrations/sqlite/2021-09-17T10:00_create_requisition_table/up.sql:6]: Looks like name and store might be the wrong way around in this sql?

```suggestion
    store_id TEXT NOT NULL REFERENCES store(id),
    name_id TEXT NOT NULL REFERENCES name(id),
```

File: repository/migrations/postgres/2021-09-20T10:00_create_invoice_table/up.sql [hunk 3]
```diff
@@ -34,4 +34,8 @@ CREATE TABLE invoice (
     shipped_datetime TIMESTAMP,
     delivered_datetime TIMESTAMP,
     verified_datetime TIMESTAMP,
-    color TEXT)
+    color TEXT,
```
Comment (clemens-msupply) [inline repository/migrations/postgres/2021-09-20T10:00_create_invoice_table/up.sql:37]: color will go into properties at some point, right?

color -> colour?

**Review by clemens-msupply — APPROVED** [review]
Looks good, see comments

File: repository/migrations/postgres/2021-09-17T10:00_create_requisition_table/up.sql (hunk 1 above)
Comment (andreievg) [reply repository/migrations/postgres/2021-09-17T10:00_create_requisition_table/up.sql:9]: Thanks James ! Nicely spotted

File: repository/migrations/sqlite/2021-09-17T10:00_create_requisition_table/up.sql (hunk 2 above)
Comment (andreievg) [reply repository/migrations/sqlite/2021-09-17T10:00_create_requisition_table/up.sql]: Thanks James ! Nicely spotted

File: repository/migrations/postgres/2021-09-20T10:00_create_invoice_table/up.sql (hunk 3 above)
Comment (andreievg) [reply repository/migrations/postgres/2021-09-20T10:00_create_invoice_table/up.sql]: > color will go into properties at some point, right?

Hmm not sure what you mean, can chat about it today

> color -> colour

efc55581891afccfdba2b75796944254537c8d71, thanks

**Pull request merged**